### PR TITLE
MFTF: Checkout With Enabled Captcha

### LINKED ISSUE #32991: [Issue] MFTF: Checkout With Enabled Captcha
This issue is automatically created based on existing pull request: magento/magento2#32986: MFTF: Checkout With Enabled Captcha

---------
This PR contains MFTF test for placing an order with enabled captcha

**Steps to reproduce:**

1 - Create customer
2 - Create product
3 - Login as customer
4- Add Product into shopping cart
5 - Place an order with incorrect captcha values
6 - Place an order with correct captcha values
7 - Perform assertions

### PR DESCRIPTION
**Pull request by DmitryTsymbal**
This PR contains MFTF test for placing an order with enabled captcha

**Steps to reproduce:**

1 - Create customer
2 - Create product
3 - Login as customer
4- Add Product into shopping cart
5 - Place an order with incorrect captcha values
6 - Place an order with correct captcha values
7 - Perform assertions

### Resolved issues:
1. [x] resolves magento/magento2#32991: MFTF: Checkout With Enabled Captcha

### CODE REVIEW

**Comment by DmitryTsymbal** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
Hi just curious! Is there any case test cover checkout without captcha

**Comment by shikhamis11** [comment]
@magento create issue

File: app/code/Magento/Captcha/Test/Mftf/Data/CaptchaConfigData.xml [hunk 1]
```diff
@@ -33,6 +33,12 @@
         <data key="label">Contact Us</data>
         <data key="value">contact_us</data>
     </entity>
+    <entity name="StorefrontCaptchaOnCheckoutConfigData">
```
Comment (shikhamis11) [inline app/code/Magento/Captcha/Test/Mftf/Data/CaptchaConfigData.xml]: @DmitryTsymbal Please add a new entity changes at the end of the file

**Review by shikhamis11 — CHANGES_REQUESTED** [review]

**Comment by Den4ik** [comment]
@magento run Functional Tests B2B, WebAPI Tests, Database Compare

File: app/code/Magento/Captcha/Test/Mftf/Data/CaptchaConfigData.xml (hunk 1 above)
Comment (DmitryTsymbal) [reply app/code/Magento/Captcha/Test/Mftf/Data/CaptchaConfigData.xml]: @shikhamis11 , thanks for noticing. I have moved new entity to the end of the file. Please check

**Comment by DmitryTsymbal** [comment]
@magento run all tests

**Review by shikhamis11 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11, thank you for the review.
[ENGCOM-9105](https://jira.corp.magento.com/browse/ENGCOM-9105) has been created to process this Pull Request
:eight_spoked_asterisk: @shikhamis11, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Review by Den4ik — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, thank you for the review.
[ENGCOM-9105](https://jira.corp.magento.com/browse/ENGCOM-9105) has been created to process this Pull Request

**Comment by mrtuvn** [comment]
https://github.com/magento/magento2/pull/33200
@DmitryTsymbal Can you help for cover for disabled case captcha ?

**Comment by DmitryTsymbal** [comment]
Hello @mrtuvn , sorry for the long reply.
As far, as I see, there are a lot of cases for checkout with disabled captcha, for example customer and guest checkout:

https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Checkout/Test/Mftf/Test/StorefrontCustomerCheckoutTest/StorefrontCustomerCheckoutTest.xml

Or

https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Checkout/Test/Mftf/Test/StorefrontGuestCheckoutTest/StorefrontGuestCheckoutTest.xml

Does it works for you, or do we need some special cases to cever?

Thanks!

**Comment by mrtuvn** [comment]
Thanks you better let do it at another seperate one

**Comment by engcom-Charlie** [comment]
:heavy_check_mark: QA Passed

**Preconditions:**

1. Create category
2. Create Product
3. Enabled the captcha at store front on payment processing page with length and symbol and in Always mode from Admin panel

**Manual testing scenario:**

1 - Login to store front 
2-  Place an order on store front
3-  Move to checkout page from mini shopping cart
4-  On second page of checkout i.e. payment page, try to submit the page with correct captcha. Customer can able to place an order.
5-  On second page of checkout i.e. payment page, try to submit the page with incorrect captcha, it should throw an error “Incorrect captcha” without further processing.
5 -  For above steps also run the StorefrontCaptchaCheckoutWithEnabledCaptchaTest  MFTF test 

**Actual Result:** :heavy_check_mark:
 
1. Customer should be able to place an order with enabled captcha.

**After:** :heavy_check_mark:  
image

image

image

image

image

image






**Tested all the manual scenarios, no impact on regression testing.**

**Review by sidolov — APPROVED** [review]

**Pull request merged**